### ISSUE TITLE
repr_table1_3dpw.yaml result reproduction issue

### ISSUE THREAD

**Issue by tcece** [sxl142/GLoT#13]
I don't know why the result of repr_table1_3dpw.yaml has not been successfully reproduced. Can you help me check why? I have run many times and it has always been this result, a very firm result.

Namespace(cfg='./configs/repr_table1_3dpw.yaml', dataset='3dpw', filter=False, frame=0, gpu='1', plot=False, render=False, render_plain=False, seq='')

Seed value for the experiment 1
GPU name -> NVIDIA GeForce RTX 2080 Ti
GPU feat -> _CudaDeviceProperties(name='NVIDIA GeForce RTX 2080 Ti', major=7, minor=5, total_memory=11019MB, multi_processor_count=68)
{'CUDNN': CfgNode({'BENCHMARK': True, 'DETERMINISTIC': False, 'ENABLED': True}),
 'DATASET': CfgNode({'SEQLEN': 16, 'OVERLAP': 0.5}),
 'DEBUG': False,
 'DEBUG_FREQ': 5,
 'DEVICE': 'cuda',
 'EXP_NAME': 'table1_3dpw_exp',
 'LOGDIR': 'experiments/18-10-2024_08-11-45_table1_3dpw_exp',
 'LOSS': {'D_MOTION_LOSS_W': 0.0,
          'KP_2D_W': 300.0,
          'KP_3D_W': 300.0,
          'POSE_W': 60.0,
          'SHAPE_W': 0.06,
          'use_accel': False,
          'vel_or_accel_2d_weight': 10.0,
          'vel_or_accel_3d_weight': 100.0},
 'MODEL': {'MODEL_NAME': 'GLoT',
           'atten_drop': 0.0,
           'd_model': 512,
           'drop_path_r': 0.2,
           'drop_reg_short': 0.25,
           'dropout': 0.1,
           'mask_ratio': 0.5,
           'n_layers': 2,
           'num_head': 8,
           'short_atten_drop': 0.0,
           'short_d_model': 256,
           'short_drop_path_r': 0.2,
           'short_dropout': 0.1,
           'short_n_layers': 3,
           'short_num_head': 8,
           'stride_short': 4},
 'NUM_WORKERS': 16,
 'OUTPUT_DIR': 'experiments',
 'SEED_VALUE': 1,
 'TITLE': 'repr_table4_3dpw_model',
 'TRAIN': {'BATCH_SIZE': 64,
           'DATASETS_2D': ['Insta'],
           'DATASETS_3D': ['ThreeDPW', 'MPII3D', 'Human36M'],
           'DATASET_EVAL': 'ThreeDPW',
           'DATA_2D_RATIO': 0.6,
           'END_EPOCH': 50,
           'GEN_LR': 0.0001,
           'GEN_MOMENTUM': 0.9,
           'GEN_OPTIM': 'Adam',
           'GEN_WD': 0.0,
           'LR_PATIENCE': 5,
           'MOT_DISCR': {'ATT': {'DROPOUT': 0.2,
                                 'LAYERS': 3,
                                 'SIZE': 1024},
                         'FEATURE_POOL': 'attention',
                         'HIDDEN_SIZE': 1024,
                         'LR': 0.0001,
                         'MOMENTUM': 0.9,
                         'NUM_LAYERS': 2,
                         'OPTIM': 'Adam',
                         'UPDATE_STEPS': 1,
                         'WD': 0.0001},
           'NUM_ITERS_PER_EPOCH': 1000,
           'OVERLAP': 0.0,
           'PRETRAINED': 'experiments/17-10-2024_15-41-15_table1_3dpw_exp/model_best.pth.tar',
           'PRETRAINED_REGRESSOR': 'data/base_data/spin_model_checkpoint.pth.tar',
           'RESUME': '',
           'START_EPOCH': 0,
           'val_epoch': 25},
 'render': False}
InstaVariety number of dataset objects 130431
3DPW Dataset overlap ratio:  0.0
Loaded 3dpw dataset from data/preprocessed_data/3dpw_train_occ_db.pt
is_train:  True
3dpw - number of dataset objects 1403
MPII3D Dataset overlap ratio:  0.0
Loaded mpii3d dataset from data/preprocessed_data/mpii3d_train_scale12_occ_db.pt
is_train:  True
mpii3d - number of dataset objects 59934
Human36M Dataset overlap ratio:  0.0
Loaded h36m dataset from data/preprocessed_data/h36m_train_25fps_occ_db.pt
is_train:  True
h36m - number of dataset objects 48456
3DPW Dataset overlap ratio:  0.9375
Loaded 3dpw dataset from data/preprocessed_data/3dpw_val_db.pt
is_train:  False
3dpw - number of dataset objects 10160
/home/u20/anaconda3/envs/glot/lib/python3.7/site-packages/torch/nn/_reduction.py:43: UserWarning: size_average and reduce args will be deprecated, please use reduction='mean' instead.
  warnings.warn(warning.format(ret))
=> loaded pretrained model from 'data/base_data/spin_model_checkpoint.pth.tar'
net: GLoT(
  (proj_short): Linear(in_features=2048, out_features=256, bias=True)
  (proj_mem): Linear(in_features=512, out_features=256, bias=True)
  (local_trans_de): CrossAttention(
    (kv_l): Linear(in_features=256, out_features=512, bias=False)
    (q_l): Linear(in_features=256, out_features=256, bias=False)
    (attn_drop): Dropout(p=0.0, inplace=False)
    (proj): Linear(in_features=256, out_features=256, bias=True)
    (proj_drop): Dropout(p=0.0, inplace=False)
  )
  (local_trans_en): Transformer(
    (blocks): ModuleList(
      (0): Block(
        (norm1): LayerNorm((256,), eps=1e-06, elementwise_affine=True)
        (attn): Attention(
          (qkv): Linear(in_features=256, out_features=768, bias=True)
          (attn_drop): Dropout(p=0.0, inplace=False)
          (proj): Linear(in_features=256, out_features=256, bias=True)
          (proj_drop): Dropout(p=0.1, inplace=False)
        )
        (drop_path): Identity()
        (norm2): LayerNorm((256,), eps=1e-06, elementwise_affine=True)
        (mlp): Mlp(
          (fc1): Linear(in_features=256, out_features=1024, bias=True)
          (act): GELU()
          (fc2): Linear(in_features=1024, out_features=256, bias=True)
          (drop): Dropout(p=0.1, inplace=False)
        )
      )
      (1): Block(
        (norm1): LayerNorm((256,), eps=1e-06, elementwise_affine=True)
        (attn): Attention(
          (qkv): Linear(in_features=256, out_features=768, bias=True)
          (attn_drop): Dropout(p=0.0, inplace=False)
          (proj): Linear(in_features=256, out_features=256, bias=True)
          (proj_drop): Dropout(p=0.1, inplace=False)
        )
        (drop_path): DropPath(drop_prob=0.100)
        (norm2): LayerNorm((256,), eps=1e-06, elementwise_affine=True)
        (mlp): Mlp(
          (fc1): Linear(in_features=256, out_features=1024, bias=True)
          (act): GELU()
          (fc2): Linear(in_features=1024, out_features=256, bias=True)
          (drop): Dropout(p=0.1, inplace=False)
        )
      )
      (2): Block(
        (norm1): LayerNorm((256,), eps=1e-06, elementwise_affine=True)
        (attn): Attention(
          (qkv): Linear(in_features=256, out_features=768, bias=True)
          (attn_drop): Dropout(p=0.0, inplace=False)
          (proj): Linear(in_features=256, out_features=256, bias=True)
          (proj_drop): Dropout(p=0.1, inplace=False)
        )
        (drop_path): DropPath(drop_prob=0.200)
        (norm2): LayerNorm((256,), eps=1e-06, elementwise_affine=True)
        (mlp): Mlp(
          (fc1): Linear(in_features=256, out_features=1024, bias=True)
          (act): GELU()
          (fc2): Linear(in_features=1024, out_features=256, bias=True)
          (drop): Dropout(p=0.1, inplace=False)
        )
      )
    )
    (norm): LayerNorm((256,), eps=1e-06, elementwise_affine=True)
  )
  (regressor): HSCR(
    (fc1): Linear(in_features=266, out_features=1024, bias=True)
    (fc2): Linear(in_features=400, out_features=1024, bias=True)
    (drop1): Dropout(p=0.25, inplace=False)
    (drop2): Dropout(p=0.25, inplace=False)
    (decshape): Linear(in_features=1024, out_features=10, bias=True)
    (deccam): Linear(in_features=2051, out_features=3, bias=True)
    (smpl): SMPL(
      Number of betas: 10
      (vertex_joint_selector): VertexJointSelector()
    )
    (local_reg): KTD(
      (joint_regs): ModuleList(
        (0): Linear(in_features=1030, out_features=6, bias=True)
        (1): Linear(in_features=1036, out_features=6, bias=True)
        (2): Linear(in_features=1036, out_features=6, bias=True)
        (3): Linear(in_features=1036, out_features=6, bias=True)
        (4): Linear(in_features=1042, out_features=6, bias=True)
        (5): Linear(in_features=1042, out_features=6, bias=True)
        (6): Linear(in_features=1042, out_features=6, bias=True)
        (7): Linear(in_features=1048, out_features=6, bias=True)
        (8): Linear(in_features=1048, out_features=6, bias=True)
        (9): Linear(in_features=1048, out_features=6, bias=True)
        (10): Linear(in_features=1054, out_features=6, bias=True)
        (11): Linear(in_features=1054, out_features=6, bias=True)
        (12): Linear(in_features=1054, out_features=6, bias=True)
        (13): Linear(in_features=1054, out_features=6, bias=True)
        (14): Linear(in_features=1054, out_features=6, bias=True)
        (15): Linear(in_features=1060, out_features=6, bias=True)
        (16): Linear(in_features=1060, out_features=6, bias=True)
        (17): Linear(in_features=1060, out_features=6, bias=True)
        (18): Linear(in_features=1066, out_features=6, bias=True)
        (19): Linear(in_features=1066, out_features=6, bias=True)
        (20): Linear(in_features=1072, out_features=6, bias=True)
        (21): Linear(in_features=1072, out_features=6, bias=True)
        (22): Linear(in_features=1078, out_features=6, bias=True)
        (23): Linear(in_features=1078, out_features=6, bias=True)
      )
      (decshape): Linear(in_features=1024, out_features=10, bias=True)
      (deccam): Linear(in_features=1024, out_features=3, bias=True)
    )
  )
  (global_modeling): GMM(
    (proj): Linear(in_features=2048, out_features=512, bias=True)
    (trans): Transformer(
      (mask_token_mlp): Sequential(
        (0): Linear(in_features=157, out_features=256, bias=True)
        (1): ReLU()
        (2): Linear(in_features=256, out_features=512, bias=True)
        (3): ReLU()
        (4): Linear(in_features=512, out_features=256, bias=True)
      )
      (blocks): ModuleList(
        (0): Block(
          (norm1): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
          (attn): Attention(
            (qkv): Linear(in_features=512, out_features=1536, bias=True)
            (attn_drop): Dropout(p=0.0, inplace=False)
            (proj): Linear(in_features=512, out_features=512, bias=True)
            (proj_drop): Dropout(p=0.1, inplace=False)
          )
          (drop_path): Identity()
          (norm2): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
          (mlp): Mlp(
            (fc1): Linear(in_features=512, out_features=2048, bias=True)
            (act): GELU()
            (fc2): Linear(in_features=2048, out_features=512, bias=True)
            (drop): Dropout(p=0.1, inplace=False)
          )
        )
        (1): Block(
          (norm1): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
          (attn): Attention(
            (qkv): Linear(in_features=512, out_features=1536, bias=True)
            (attn_drop): Dropout(p=0.0, inplace=False)
            (proj): Linear(in_features=512, out_features=512, bias=True)
            (proj_drop): Dropout(p=0.1, inplace=False)
          )
          (drop_path): DropPath(drop_prob=0.200)
          (norm2): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
          (mlp): Mlp(
            (fc1): Linear(in_features=512, out_features=2048, bias=True)
            (act): GELU()
            (fc2): Linear(in_features=2048, out_features=512, bias=True)
            (drop): Dropout(p=0.1, inplace=False)
          )
        )
      )
      (norm): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (decoder_blocks): ModuleList(
        (0): Block(
          (norm1): LayerNorm((256,), eps=1e-06, elementwise_affine=True)
          (attn): Attention(
            (qkv): Linear(in_features=256, out_features=768, bias=True)
            (attn_drop): Dropout(p=0.0, inplace=False)
            (proj): Linear(in_features=256, out_features=256, bias=True)
            (proj_drop): Dropout(p=0.1, inplace=False)
          )
          (drop_path): Identity()
          (norm2): LayerNorm((256,), eps=1e-06, elementwise_affine=True)
          (mlp): Mlp(
            (fc1): Linear(in_features=256, out_features=1024, bias=True)
            (act): GELU()
            (fc2): Linear(in_features=1024, out_features=256, bias=True)
            (drop): Dropout(p=0.1, inplace=False)
          )
        )
      )
      (decoder_embed): Linear(in_features=512, out_features=256, bias=True)
      (decoder_norm): LayerNorm((256,), eps=1e-06, elementwise_affine=True)
    )
    (out_proj): Linear(in_features=256, out_features=2048, bias=True)
    (regressor): Regressor(
      (fc1): Linear(in_features=2205, out_features=1024, bias=True)
      (drop1): Dropout(p=0.5, inplace=False)
      (fc2): Linear(in_features=1024, out_features=1024, bias=True)
      (drop2): Dropout(p=0.5, inplace=False)
      (decpose): Linear(in_features=1024, out_features=144, bias=True)
      (decshape): Linear(in_features=1024, out_features=10, bias=True)
      (deccam): Linear(in_features=1024, out_features=3, bias=True)
      (smpl): SMPL(
        Number of betas: 10
        (vertex_joint_selector): VertexJointSelector()
      )
    )
  )
)
params num: 16754444
=> no checkpoint found at ''
Epoch 1/50 |                 
![image](https://github.com/user-attachments/assets/f23c7289-dbb0-4c8a-a850-4b6e6c5159dd)

**Comment by tcece** [comment]
![image](https://github.com/user-attachments/assets/1ff60a04-0b74-4792-9968-9c627f892cad)
